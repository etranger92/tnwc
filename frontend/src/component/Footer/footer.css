@import '../../style/share.css';

#footer {
    margin-top: var(--margin-section-y);
    padding-left: var(--padding-x);
    padding-right: var(--padding-x);
    padding-top: var(--padding-y);
    padding-bottom: var(--padding-y);
    background-color: var(--txt-color);
    min-height: 80vh;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 3rem;
    padding-bottom: 0rem;
}

#footer p {
    color: rgba(255, 255, 255, 0.884);

}



#footer-contact {
    margin-top: 1rem;
}

#footer-contact svg path {
    fill: white;
}

#footer-container-one {
    display: flex;
    flex-wrap: wrap;
    flex: 2 1;
    border-radius: 10px 10px 10px 10px;
    padding: 1rem;
    /* box-shadow: 4.5px 5.4px 10px yellow; */
    box-shadow: 0px 0px 10px 0px white;
}

@media only screen and (min-width: 750px) {

    #footer-container-one {
        padding: 3rem;
    }
}

#footer-container-one p {
    text-align: start;
}

.footer_sub_container {
    width: 100%;
}

.footer_sub_container:first-child h2 {
    margin-top: 0rem;
}


.footer_sub_container h2 {
    margin-top: 2rem;
    margin-bottom: 0.5rem;
    color: white;

}


@media screen and (min-width: 1000px) {
    .footer_sub_container {
        width: 25%;
        padding-left: 1rem;
        padding-right: 1rem;
    }

    .footer_sub_container h2 {
        margin-top: 0;

    }

    #footer-container-one>div:nth-child(3) {
        width: 17%;
    }

    #footer-container-one>div:nth-child(4) {
        width: 33%;
    }
}


.footer_sub_container:last-child ul {
    padding-right: 0.2rem;
}

#footer-contact li {
    display: flex;
    align-items: center;
}

#footer-contact li p {
    padding-left: 0.3rem;
}

#footer-container-one>div:nth-child(1) {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
}

#footer-container-one>div:nth-child(1) h2 {
    width: 100%;
}

.footer_course {
    display: flex;
    align-items: center;
    width: 50%;
    height: 10vh;
    margin-bottom: .5rem;
}

.footer_course p {
    flex: 1;
    text-align: center;
    padding-left: .4rem;
}


.footer_course_img {
    height: 100%;
    flex: 1;
}

.footer_course img {
    width: 100%;
    height: 100%;
    border-radius: 5px 5px 5px 5px;
}

#footer-container-two {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 1rem;
    padding-bottom: 1.5rem;
}

#footer-container-two a {
    color: var(--fourth-color) !important;
}

#footer-container-two>* {
    padding-top: 0.6rem;
}

#footer-container-two p {
    padding-left: var(--first-padding);
    color: white !important;
}

#footer-container-two::before {
    content: "";
    border: 1px solid var(--primary-color);
    width: 80%;
    top: 0;
    margin-top: 1rem;
    margin-bottom: 1rem;
}

#footer-container-two ul {
    display: flex;
}

#footer-container-two ul li {
    padding-left: 1rem;
    padding-right: 1rem;
}

#footer #nav-bar>li:nth-child(3)>a>svg path {
    fill: white;
}

#footer-container-one>div:nth-child(1) {
    width: 60vw;
}

@media screen and (min-width: 1000px) {
    #footer-container-one>div:nth-child(1) {
        width: 25%;

    }

    #footer_container {
        margin-bottom: 0rem;
    }
}

#footer-container-two ul:first-child {
    margin-top: 2rem;
}

#footer-container-two>p {
    padding-top: 0.3rem;
}