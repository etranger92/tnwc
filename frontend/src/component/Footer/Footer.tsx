/* eslint-disable react/no-unescaped-entities */
import React from "react";
import IconFacebook from "../ElementsHTML/Icons/iconFacebook";
import IconInstagram from "../ElementsHTML/Icons/iconInstagram";
import IconLinkedin from "../ElementsHTML/Icons/iconLinkedin";
import IconLocation from "../ElementsHTML/Icons/iconLocation";
import IconMail from "../ElementsHTML/Icons/iconMail";
import IconPhone from "../ElementsHTML/Icons/iconPhone";
import IconTwitter from "../ElementsHTML/Icons/iconTwitter";
import NavBar from "../Navigation/NavBar/NavBar";
import aet from "../../image/course/aet.png";
import cet from "../../image/course/cet.jpg";
import det from "../../image/course/det.jpg";
import qtls from "../../image/course/qtls.jpg";
import "./footer.css";

const Footer: React.FC = () => {
  return (
    <div id="footer">
      <div id="footer-container-one">
        <div className="footer_sub_container">
          <h2>Courses</h2>{" "}
          <div className="footer_course">
            <div className="footer_course_img">
              {" "}
              <img src={aet} alt="education and training" />
            </div>
            <p>AET</p>
          </div>
          <div className="footer_course">
            <div className="footer_course_img">
              {" "}
              <img src={cet} alt="education and training" />
            </div>
            <p>CET</p>
          </div>
          <div className="footer_course">
            <div className="footer_course_img">
              {" "}
              <img src={det} alt="education and training" />
            </div>
            <p>DET</p>
          </div>
          <div className="footer_course">
            <div className="footer_course_img">
              {" "}
              <img src={qtls} alt="teacher status" />
            </div>
            <p>QTLS</p>
          </div>
        </div>
        <div className="footer_sub_container">
          <h2>About</h2>{" "}
          <p>
            The N.W.C is a leading 21st century Awarding Organisation founded to
            meet the developing education and training needs of learners and
            employers from the UK and around the globe.
          </p>
        </div>
        <div className="footer_sub_container">
          <h2>Follow us</h2>
          <p>Coming soon</p>
        </div>
        <div className="footer_sub_container">
          {" "}
          <h2>Find us </h2>
          <p>Feel free to contact us and we will be back at you. </p>
          <ul id="footer-contact">
            <li>
              <IconLocation /> <p>England</p>
            </li>
            <li>
              {" "}
              <IconPhone />
              <p>07751593970</p>
            </li>
            <li>
              {" "}
              <IconMail />
              <p>admin@thenorthwestcollege.co.uk</p>
            </li>
          </ul>
        </div>
      </div>
      <div id="footer-container-two">
        <NavBar isItClicked={true} />
        <p>Copyright @2020 | Designed by N.S</p>
        <ul>
          <li>
            <IconFacebook />
          </li>
          <li>
            <IconTwitter />
          </li>
          <li>
            <IconInstagram />
          </li>
          <li>
            <IconLinkedin />
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Footer;
