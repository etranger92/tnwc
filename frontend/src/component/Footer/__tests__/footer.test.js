import * as React from 'react';
import Footer from "../Footer.tsx"
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("We test the innerHTML of the Footer", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });
  test("Snapshot footer", () => {
    //It works
 expect(toJson(wrapper)).toMatchSnapshot();
});
});
