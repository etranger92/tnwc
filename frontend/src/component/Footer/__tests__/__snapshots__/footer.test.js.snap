// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We test the innerHTML of the Footer Snapshot footer 1`] = `
<div
  id="footer"
>
  <div
    id="footer-container-one"
  >
    <div
      className="footer_sub_container"
    >
      <h2>
        Courses
      </h2>
       
      <div
        className="footer_course"
      >
        <div
          className="footer_course_img"
        >
           
          <img
            alt="education and training"
            src="aet.png"
          />
        </div>
        <p>
          AET
        </p>
      </div>
      <div
        className="footer_course"
      >
        <div
          className="footer_course_img"
        >
           
          <img
            alt="education and training"
            src="cet.jpg"
          />
        </div>
        <p>
          CET
        </p>
      </div>
      <div
        className="footer_course"
      >
        <div
          className="footer_course_img"
        >
           
          <img
            alt="education and training"
            src="det.jpg"
          />
        </div>
        <p>
          DET
        </p>
      </div>
      <div
        className="footer_course"
      >
        <div
          className="footer_course_img"
        >
           
          <img
            alt="teacher status"
            src="qtls.jpg"
          />
        </div>
        <p>
          QTLS
        </p>
      </div>
    </div>
    <div
      className="footer_sub_container"
    >
      <h2>
        About
      </h2>
       
      <p>
        The N.W.C is a leading 21st century Awarding Organisation founded to meet the developing education and training needs of learners and employers from the UK and around the globe.
      </p>
    </div>
    <div
      className="footer_sub_container"
    >
      <h2>
        Follow us
      </h2>
      <p>
        Coming soon
      </p>
    </div>
    <div
      className="footer_sub_container"
    >
       
      <h2>
        Find us 
      </h2>
      <p>
        Feel free to contact us and we will be back at you. 
      </p>
      <ul
        id="footer-contact"
      >
        <li>
          <IconLocation />
           
          <p>
            England
          </p>
        </li>
        <li>
           
          <IconPhone />
          <p>
            07751593970
          </p>
        </li>
        <li>
           
          <IconMail />
          <p>
            admin@thenorthwestcollege.co.uk
          </p>
        </li>
      </ul>
    </div>
  </div>
  <div
    id="footer-container-two"
  >
    <NavBar
      isItClicked={true}
    />
    <p>
      Copyright @2020 | Designed by N.S
    </p>
    <ul>
      <li>
        <IconFacebook />
      </li>
      <li>
        <IconTwitter />
      </li>
      <li>
        <IconInstagram />
      </li>
      <li>
        <IconLinkedin />
      </li>
    </ul>
  </div>
</div>
`;
